### PR TITLE
External call hyper parameter checks

### LINKED ISSUE #100: External call hyper-parameter checks
(From #72)

External calls may optionally have hyper-parameters specified.

```swift
call! ext.someFunc() // no hyper-parameters
call(value: Wei(100))! ext.somePayableFunc() // value hyper-parameter
call(gas: 200)! ext.someFunc() // gas hyper-parameter
call(value: Wei(100), gas: 200)! ext.somePayableFunc() // both
```

(The hyper-parameters may be specified for all `call` modes, i.e. default `call`, forced `call!`, value-returning `call?`.)

 - [ ] check that only `value` and `gas` are ever specified
 - [ ] check that neither `value` nor `gas` is duplicated

### PR DESCRIPTION
**Pull request by Aurel300**
Closes #100.

## Notes

The test will need to be adjusted after #104 and other external call tickets.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -26,7 +26,8 @@ test-nogen: lint release
 	swift run -c release lite
 
 lint:
-	swiftlint lint --strict
+	#swiftlint lint --strict
```
Comment (nvgrw) [inline Makefile]: no

File: Tests/Integration/BehaviorTests/run_behavior_tests.sh [hunk 2]
```diff
@@ -3,7 +3,7 @@
 NUM_FAILED=0
 
 echo "Compiling Flint sources"
-./compile_behavior_tests.sh
+#./compile_behavior_tests.sh
```
Comment (nvgrw) [inline Tests/Integration/BehaviorTests/run_behavior_tests.sh]: This is not how you pass CI

File: Sources/SemanticAnalyzer/SemanticError.swift [hunk 3]
```diff
@@ -478,8 +478,23 @@ extension Diagnostic {
 
   static func publicAndVisible(_ variableDeclaration: VariableDeclaration) -> Diagnostic {
     let identName = variableDeclaration.identifier.name
-   return Diagnostic(severity: .error, sourceLocation: variableDeclaration.sourceLocation,
-                     message: "Cannot declare variable '\(identName)' both public and visible")
+    return Diagnostic(severity: .error, sourceLocation: variableDeclaration.sourceLocation,
+                      message: "Cannot declare variable '\(identName)' both public and visible")
+  }
+
+  static func invalidExternalCallHyperParameter(_ identifier: Identifier) -> Diagnostic {
```
Comment (nvgrw) [inline Sources/SemanticAnalyzer/SemanticError.swift]: Call this `invalidExternalCallConfigurationParameter` + the others or rename occurrences of `ConfigurationParameter` in the codebase for consistency

File: Tests/Integration/SemanticTests/external_calls.flint [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+  
+  public init() {
+    call! self.someFunc()
+    call(gas: 100)! self.someFunc()
+    call(value: Wei(100))! self.someFunc()
+    call(value: Wei(100), gas: 100)! self.someFunc()
+
+    call(gas: 100, value: Wei(200), gas: 300)! self.someFunc() // expected-error {{'gas' hyper-parameter was already specified}}
+    call(value: Wei(100), value: Wei(200), value: Wei(300))! self.someFunc()
+    // expected-error@-1 {{'value' hyper-parameter was already specified}}
```
Comment (nvgrw) [inline Tests/Integration/SemanticTests/external_calls.flint]: Can you put this on the same line as the call that you're asserting for

File: Tests/Integration/SemanticTests/external_calls.flint [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    // expected-error@-2 {{'value' hyper-parameter was already specified}}
```
Comment (nvgrw) [inline Tests/Integration/SemanticTests/external_calls.flint]: Can you indent this and change -2 to -1 in line with the previous comment

File: Sources/SemanticAnalyzer/SemanticError.swift (hunk 3 above)
Comment (Aurel300) [reply Sources/SemanticAnalyzer/SemanticError.swift]: Changed `configurationParameter` to `hyperParameter` throughout codebase.

Comment (nvgrw) [reply Sources/SemanticAnalyzer/SemanticError.swift]: anise

**Pull request merged**